solarforecastarbiter/reports/figures/plotly_figures.py

```diff
@@ -0,0 +1,432 @@
+
+    x_axis_label = human_category
+    y_axis_label = metric_name
+
+    data = df[(df['category'] == category) & (df['metric'] == metric)]
+
+    x_offset = None
+
+    # Special handling for x-axis with dates
+    if category == 'weekday':
+        x_ticks = calendar.day_abbr[0:]
+
+    elif category == 'hour':
+        x_ticks = list(range(25))
+        # plotly's offset of 0, makes the bars left justified at the tick
+        x_offset = 0
+    else:
+        x_ticks = None
+
+    y_data = np.asarray(data['value'])
+    if len(y_data) == 0:
+        y_range = (None, None)
+    else:
+        y_min = np.nanmin(y_data)
+        y_max = np.nanmax(y_data)
+        y_range = calc_y_start_end(y_min, y_max)
+
+    unique_names = np.unique(np.asarray(data['name']))
+    palette = [next(palette) for _ in unique_names]
+    for i, name in enumerate(unique_names):
+        plot_data = data[data['name'] == name]
+        if len(plot_data['index']):
+            x_values = plot_data['index']
+        else:
+            x_values = []
+        if category == 'weekday':
+            # Fill with mon-fri values and pass to enforce displaying the full
+            # week of data.
+            y_values = [plot_data[plot_data['index'] == day]['value'].iloc[0]
+                        if not plot_data[plot_data['index'] == day].empty
+                        else 0 for day in x_ticks]
+            x_values = x_ticks
+        else:
+            y_values = plot_data['value']
+        # Create figure
+        title = name + ' ' + metric_name
+        fig = go.Figure()
+        fig.add_trace(go.Bar(x=x_values, y=y_values, offset=x_offset,
+                             marker=go.bar.Marker(color=palette[i])))
+
+        fig.update_layout(
+            title=f'<b>{title}</b>',
+            xaxis_title=x_axis_label,
+            yaxis_title=y_axis_label,
+            **PLOT_LAYOUT_DEFAULTS)
+        configure_axes(fig, x_ticks, y_range)
+        figs[name] = fig
+    return figs
+
+
+def nested_bar():
+    raise NotImplementedError
+
+
+def joint_distribution():
+    raise NotImplementedError
+
+
+def marginal_distribution():
+    raise NotImplementedError
+
+
+def taylor_diagram():
+    raise NotImplementedError
+
+
+def probabilistic_timeseries():
+    raise NotImplementedError
+
+
+def reliability_diagram():
+    raise NotImplementedError
+
+
+def rank_histogram():
+    raise NotImplementedError
+
+
+@contextmanager
+def _make_webdriver():
+    """Necessary until Bokeh 2.0 when using chrome/firefox drivers will be
+    preferred and to avoid zombie phantomjs processes for now"""
+    from selenium import webdriver
+    with warnings.catch_warnings():
+        warnings.simplefilter('ignore')
+        try:
+            driver = webdriver.PhantomJS()
+        except Exception:
+            yield None
+        else:
+            yield driver
+            driver.quit()
+
+
+def output_svg(fig):
+    """
+    Generates an SVG from the Bokeh or Plotly figure. Errors in the
```

Comment:
only works for plotly. Add note that it requires orca and maybe link to plotly docs for install instructions